src/com/oltpbenchmark/api/Worker.java

```diff
@@ -402,11 +402,8 @@ public final void run() {
                 }
             }
             // In case of transaction failures, the end times will not be populated.
-            if (executionState.getEndOperation() == 0) {
-              executionState.setEndOperation(System.nanoTime());
-            }
-            if (executionState.getEndConnection() == 0) {
-              executionState.setEndConnection(System.nanoTime());
+            if (executionState.getEndOperation() == 0 || executionState.getEndConnection() == 0) {
+              continue work;
```

Comment:
it seems like the behavior now will be that we will retry indefinitely. should we retry N times before calling this a failure?

Reply:
Done.